flink/v1.16/flink/src/test/java/org/apache/iceberg/flink/DataGenerators.java

```diff
@@ -0,0 +1,746 @@
+                "struct_of_struct",
+                Types.StructType.of(
+                    required(101, "id", Types.IntegerType.get()),
+                    required(
+                        102,
+                        "person_struct",
+                        Types.StructType.of(
+                            Types.NestedField.required(201, "name", Types.StringType.get()),
+                            Types.NestedField.required(202, "address", Types.StringType.get()))))));
+
+    private final RowType flinkRowType = FlinkSchemaUtil.convert(icebergSchema);
+
+    @Override
+    public Schema icebergSchema() {
+      return icebergSchema;
+    }
+
+    @Override
+    public RowType flinkRowType() {
+      return flinkRowType;
+    }
+
+    @Override
+    public GenericRecord generateIcebergGenericRecord() {
+      Schema structSchema =
+          new Schema(icebergSchema.findField("struct_of_struct").type().asStructType().fields());
+      Schema personSchema =
+          new Schema(structSchema.findField("person_struct").type().asStructType().fields());
+      GenericRecord person = GenericRecord.create(personSchema);
+      person.setField("name", "Jane");
+      person.setField("address", "Apple Park");
+      GenericRecord struct = GenericRecord.create(structSchema);
+      struct.setField("id", 1);
+      struct.setField("person_struct", person);
+      GenericRecord genericRecord = GenericRecord.create(icebergSchema);
+      genericRecord.setField("partition_field", "partition_value");
+      genericRecord.setField("struct_of_struct", struct);
+      return genericRecord;
+    }
+
+    @Override
+    public RowData generateFlinkRowData() {
+      return GenericRowData.of(
+          StringData.fromString("partition_value"),
+          GenericRowData.of(
+              1,
+              GenericRowData.of(
+                  StringData.fromString("Jane"), StringData.fromString("Apple Park"))));
+    }
+  }
+
+  public static class ArrayOfPrimitive implements DataGenerator {
+    private final Schema icebergSchema =
+        new Schema(
+            Types.NestedField.required(1, "partition_field", Types.StringType.get()),
+            Types.NestedField.required(
+                2, "array_of_int", Types.ListType.ofRequired(101, Types.IntegerType.get())));
+
+    private final RowType flinkRowType = FlinkSchemaUtil.convert(icebergSchema);
+
+    @Override
+    public Schema icebergSchema() {
+      return icebergSchema;
+    }
+
+    @Override
+    public RowType flinkRowType() {
+      return flinkRowType;
+    }
+
+    @Override
+    public GenericRecord generateIcebergGenericRecord() {
+      GenericRecord genericRecord = GenericRecord.create(icebergSchema);
+      genericRecord.setField("partition_field", "partition_value");
+      genericRecord.setField("array_of_int", Arrays.asList(1, 2, 3));
+      return genericRecord;
+    }
+
+    @Override
+    public RowData generateFlinkRowData() {
+      Integer[] arr = {1, 2, 3};
```

Comment:
`arr` seems to be class var, `generateIcebergGenericRecord` can also use it?

Reply:
`generateIcebergGenericRecord` needs a list as `Arrays.asList(1, 2, 3)` . so it is not reusable.